googlecloudstorage_blueprints/remove_file.py

```diff
@@ -0,0 +1,170 @@
+
+
+def set_environment_variables(args):
+    """
+    Set GCP credentials as environment variables if they're provided via keyword
+    arguments rather than seeded as environment variables. This will override
+    system defaults.
+    """
+    credentials = args.gcp_application_credentials
+    try:
+        json_credentials = json.loads(credentials)
+        fd, path = tempfile.mkstemp()
+        print(f'Storing json credentials temporarily at {path}')
+        with os.fdopen(fd, 'w') as tmp:
+            tmp.write(credentials)
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = path
+        return path
+    except Exception:
+        print('Using specified json credentials file')
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = credentials
+        return
+
+
+def find_google_cloud_storage_file_names(bucket, prefix=''):
+    """
+    Fetched all the files in the bucket which are returned in a list as
+    Google Blob objects
+    """
+    return list(bucket.list_blobs(prefix=prefix))
+
+
+
+def get_gclient(args):
+    """
+    Attempts to create the Google Cloud Storage Client with the associated
+    environment variables
+    """
+    try:
+        gclient = storage.Client()
+    except Exception:
+        print(f'Error accessing Google Cloud Storage with service account '
+              f'{args.gcp_application_credentials}')
+        sys.exit(ec.EXIT_CODE_INVALID_CREDENTIALS)
+
+    return gclient
+
+
+def get_bucket(*,
+               gclient,
+               bucket_name):
+    """
+    Fetches and returns the bucket from Google Cloud Storage
+    """
+    try:
+        bucket = gclient.get_bucket(bucket_name)
+    except NotFound as e:
+        print(f'Bucket {bucket_name} does not exist\n {e}')
+        sys.exit(ec.EXIT_CODE_INVALID_BUCKET)
+
+    return bucket
+
+
+def get_storage_blob(bucket, source_folder_name, source_file_name):
+    """
+    Fetches and returns the single source file blob from the buck on
+    Google Cloud Storage
+    """
+    source_path = source_file_name
+    if source_folder_name != '':
+        source_path = f'{source_folder_name}/{source_file_name}'
+    blob = bucket.get_blob(source_path)
+    try:
+        blob.exists()
+        return blob
+    except Exception as e:
+        print(f'File {source_path} does not exist')
+        sys.exit(ec.EXIT_CODE_FILE_NOT_FOUND)
+
+
+
+def delete_google_cloud_storage_file(blob):
+    """
+    Deletes a selected file from Google cloud storage
+    """
+    blob_bucket, blob_name = blob.bucket.name, blob.name
+    blob.delete()
+    print(f"Blob {blob_bucket}/{blob_name} delete ran successfully")
+
+
+def gcp_find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob)
+
+    return matching_file_names
```

Comment:
For sustainability, I'd prefer to remove reference to this function and use files.find_all_file_matches() from shipyard_utils. It does exactly the same thing, with the addition of printing the number of matches and their names (which means some of your current prints might need to be rearranged).

Want to centralize this so if the logic ever has to be updated, it gets updated in every Blueprint.